arch/risc-v/include/setjmp.h

```diff
@@ -0,0 +1,54 @@
+/****************************************************************************
+ * arch/risc-v/include/setjmp.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __ARCH_RISCV_INCLUDE_SETJUMP_H
+#define __ARCH_RISCV_INCLUDE_SETJUMP_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/types.h>
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+struct setjmp_buf_s
+{
+  /* Note: size_t corresponds to the actual register width: 32 or 64 bit */
+
+  /* TODO: float register context */
+
+  size_t regs[14];
```

Comment:
We need to make sure this is aligned on an 8 byte boundary or we will have issues with 32bit when fp is enabled.

I don't think this is correct, and we should be making sure we are on a 8-byte boundary otherwise 32bit fp will be miss aligned. Can we instead use `long long` as the base type (8bytes for RV64 and RV32) for the array and then compute the length.

long long regs[((14 * sizeof(long) + 12 * sizeof(double)) / sizeof(long long)]

I'm basing this on the calling convention docs in the spec https://riscv.org/wp-content/uploads/2015/01/riscv-calling.pdf

We do not have support for it yet, but I would expect to see RV32E which I think would be 4 instead of 12 and no FP
http://www.five-embeddev.com/riscv-isa-manual/latest/rv32e.html

Reply:
Alternatively we punt on the floating point alignment for this PR, but in that case `long` is still the correct type to be using here for both 32 and 64 bit.

Reply:
Thanks for suggestion, now NuttX don't support float register (both f/d instruction) for risc-v or rv32e based platform

Reply:
in that case we should just go with `long regs[14];`  (just changing you original from size_t to long) so there is no confusion about what is supported.